### PR TITLE
chore: update the icu-72.1 patch file

### PR DESCRIPTION
**Pull request by realtimetodie**
Updates the libicu v72.1 patch file and removes all remaining Bazel build artifacts (for example WORKSPACE and BUILD.bazel files) from the icu source code., to prevent Bazel from treating the foreign library as a Bazel project.

Related

- https://github.com/envoyproxy/envoy/issues/26395

### CODE REVIEW

**Comment by realtimetodie** [comment]
Oups I forgot to rebase from upstream/main, force-pushed again

File: bazel/repository_locations.bzl [hunk 1]
```diff
@@ -378,6 +378,10 @@ REPOSITORY_LOCATIONS_SPEC = dict(
         project_name = "ICU Library",
         project_desc = "Development files for International Components for Unicode",
         project_url = "https://github.com/unicode-org/icu",
+        # When this is updated, make sure to update the icu.patch patch file and remove
+        # all remaining Bazel build artifacts (for example WORKSPACE and BUILD.bazel files)
+        # from the icu source code, to prevent Bazel from treating the foreign library
+        # as a Bazel project.
```
Comment (phlax) [inline bazel/repository_locations.bzl:384]: could you add a link to the issue i posted (ill change the title)

**Comment by realtimetodie** [comment]
Wow I totally forgot this DCO American shyster crap, it sucks

force-pushed again

**Comment by realtimetodie** [comment]
@htuch Ready for review

**Comment by htuch** [comment]
Can you remind me why we don't want to use the external Bazel build? Generally if a dependency has a Bazel build system, we would make use of it.

**Comment by phlax** [comment]
@htuch see conversation here https://github.com/envoyproxy/envoy/issues/26395#issuecomment-1486466812

**Review by htuch — APPROVED** [review]
LGTM, thanks!

**Comment by realtimetodie** [comment]
@htuch The envoy-presubmit checks need to be triggered

**Comment by phlax** [comment]
@realtimetodie can you merge main please - we have updated the required checks so best rerun

**Pull request merged**